homeassistant/components/swiss_public_transport/coordinator.py

```diff
@@ -58,41 +70,35 @@ async def _async_update_data(self) -> DataConnection:
             )
             raise UpdateFailed from e
 
-        departure_time = (
-            dt_util.parse_datetime(self._opendata.connections[0]["departure"])
-            if self._opendata.connections[0] is not None
-            else None
-        )
-        next_departure_time = (
-            dt_util.parse_datetime(self._opendata.connections[1]["departure"])
-            if self._opendata.connections[1] is not None
-            else None
-        )
-        next_on_departure_time = (
-            dt_util.parse_datetime(self._opendata.connections[2]["departure"])
-            if self._opendata.connections[2] is not None
-            else None
-        )
-
-        if departure_time:
-            remaining_time = departure_time - dt_util.as_local(dt_util.utcnow())
-        else:
-            remaining_time = None
-
-        return DataConnection(
-            departure=departure_time,
-            next_departure=next_departure_time.isoformat()
-            if next_departure_time is not None
-            else None,
-            next_on_departure=next_on_departure_time.isoformat()
-            if next_on_departure_time is not None
-            else None,
-            train_number=self._opendata.connections[0]["number"],
-            platform=self._opendata.connections[0]["platform"],
-            transfers=self._opendata.connections[0]["transfers"],
-            duration=self._opendata.connections[0]["duration"],
-            start=self._opendata.from_name,
-            destination=self._opendata.to_name,
-            remaining_time=f"{remaining_time}",
-            delay=self._opendata.connections[0]["delay"],
-        )
+        return [
+            DataConnection(
+                departure=dt_util.parse_datetime(
+                    self._opendata.connections[i]["departure"]
+                ),
+                next_departure=dt_util.parse_datetime(
+                    self._opendata.connections[i + 1]["departure"]
+                )
+                if len(self._opendata.connections) > i + 1
+                and self._opendata.connections[i + 1] is not None
+                else None,
```

Comment:
Same

Reply:
Why are multiline if statements bad?

Reply:
I pushed a proposition with the `if` statement on one line. If you mean something else I would be glad for an example or a guideline